docs/design/features/globalization-icu.md

```diff
@@ -0,0 +1,131 @@
+
+Applications can enable NLS mode by either of the following:
+
+1. in project file:
+
+```xml
+<ItemGroup>
+  <RuntimeHostConfigurationOption Include="System.Globalization.UseNls" Value="true" />
+</ItemGroup>
+```
+
+2. in `runtimeconfig.json` file:
+
+```json
+{
+  "runtimeOptions": {
+     "configProperties": {
+       "System.Globalization.UseNls": true
+      }
+  }
+}
+```
+
+3. setting environment variable value `DOTNET_SYSTEM_GLOBALIZATION_USENLS` to `true` or `1`.
+
+*Note: value set in project file or `runtimeconfig.json` has higher priority than the environment variable.*
+
+## App-local ICU
+
+In order to provide Common Locale Data Repository (CLDR) and behavior customization, we introduced another feature for applications to be able to carry their own copy of ICU as part of their closure. 
+
+Applications can enable NLS mode by either of the following:
+
+1. in project file:
+
+```xml
+<ItemGroup>
+  <RuntimeHostConfigurationOption Include="System.Globalization.AppLocalIcu" Value="<suffix>:<version> or <version>" />
+</ItemGroup>
+```
+
+2. in `runtimeconfig.json` file:
+
+```json
+{
+  "runtimeOptions": {
+     "configProperties": {
+       "System.Globalization.AppLocalIcu": "<suffix>:<version> or <version>"
+      }
+  }
+}
+```
+
+3. setting environment variable value `DOTNET_SYSTEM_GLOBALIZATION_APPLOCALICU` to `<suffix>:<version>` or `<version>`. 
+
+`<suffix>`: this is optional in the config property, but this follows the public ICU packaging conventions as when building a custom ICU you can customize it to produce the lib names and exported symbol names to contain a suffix. i.e: `libicuucmyapp` where `myapp` is the suffix. This can't be greater than 35 chars in length for the config switch.
+`<version>`: this has to be a valid ICU version, i.e: 67.1. This version will be used to load the binaries and to get the exported symbols.
+
+To load ICU when the app-local switch is set, we use `NativeLibrary.TryLoad` api which does probing in different paths, first it tries to find the library in `NATIVE_DLL_SEARCH_DIRECTORIES` property which is created by the dotnet host based on the `deps.json` file for the app. More details [here](https://docs.microsoft.com/en-us/dotnet/core/dependency-loading/default-probing)
+
+For self contained apps, the user doesn't really need to do anything special, other than making sure ICU is side by side in the APP directory, this is because for self-contained apps, it's work directory is by default in `NATIVE_DLL_SEARCH_DIRECTORIES`. 
+
+If you're consuming ICU via a NuGet package, this will work in framework-dependent apps as NuGet will resolve the native assets and include them in the `deps.json` file and in the output directory for the app under the `runtimes` dir, then we will load it from there.
```

Comment:
```suggestion
If you're consuming ICU via a NuGet package, this will work in framework-dependent apps as NuGet will resolve the native assets and include them in the `deps.json` file and in the output directory for the app under the `runtimes` dir, and .NET will load it from there.
```

Reply:
Does `APP directory` (lower case?) mean "next to the app.dll" or "in the app.dll folder or below"?